File: lib/src/main/java/com/github/goto1134/structurizr/export/d2/model/D2WrappedStringProperty.java
package com.github.goto1134.structurizr.export.d2.model;

import com.structurizr.export.IndentingWriter;

public class D2WrappedStringProperty<K> extends D2Property<K, String> {

    public D2WrappedStringProperty(K keyword, String value) {
        super(keyword, value);
    }

    public void write(IndentingWriter writer) {
        writer.writeLine(String.format("%s: \"%s\"", keyword, value));
    }
}

File: lib/src/main/kotlin/io/github/goto1134/structurizr/export/d2/model/D2WrappedStringProperty.kt
package io.github.goto1134.structurizr.export.d2.model

import com.structurizr.export.IndentingWriter

class D2WrappedStringProperty<K>(keyword: K, value: String) : D2Property<K, String>(keyword, value) {
    override fun write(writer: IndentingWriter) {
        writer.writeLine("$keyword: \"$value\"")
    }
}
